packages/phasor/src/elements/shape/types.ts

```diff
@@ -0,0 +1,43 @@
+import type { IBound } from '../../consts.js';
+import type { HitTestOptions } from '../base-element.js';
+
+export type ShapeType =
+  | 'rect'
+  | 'triangle'
+  | 'ellipse'
+  | 'diamond'
+  | 'roundedRect';
+
+export interface ShapeRenderConfig {
+  width: number;
+  height: number;
+  strokeWidth: number;
+  fillColor?: string;
+  strokeColor?: string;
+  strokeStyle?: 'solid';
+}
+
+interface RenderSequenceStroke {
```

Comment:
I'm not sure if we should switch to command-driven render at the moment. Looks better to revert if this can't bring justifiable performance gain.

Reply:
I'm not sure too. I read the origin code and find `Path2D` cache. So I thought you design this for a purpose. In order to make cache logic consistent, I create `RenderSequence`.